packages/app/cypress/e2e/cypress-in-cypress-component.cy.ts

```diff
@@ -145,4 +145,67 @@ describe('Cypress In Cypress CT', { viewportWidth: 1500, defaultCommandTimeout:
       )
     })
   })
+
+  it('shows automation disconnected warning', () => {
+    cy.visitApp()
+    cy.get('[data-cy="spec-item"]').first().click()
+    // Let runner stabilize
+    cy.get('[data-cy="reporter-panel"]').should('be.visible')
+
+    cy.withCtx((ctx) => {
+      ctx.coreData.servers.appSocketServer?.emit('automation:disconnected')
+    })
+
+    cy.contains('h3', 'The Cypress extension has disconnected')
+
+    cy.withCtx((ctx, { sinon }) => {
+      sinon.stub(ctx.actions.project, 'launchProject').resolves()
+    })
+
+    cy.contains('button', 'Reload the browser').click()
+
+    cy.withCtx((ctx) => {
+      expect(ctx.actions.project.launchProject).to.have.been.called
+    })
+  })
+
+  it('shows automation missing warning', () => {
+    let connectedCallback: any
+
+    cy.visitApp()
+
+    cy.window().then((win) => {
+      const originalEmit: Function = win.ws?.emit || function () {}
+      const stub = cy.stub(win.ws as any, 'emit')
+
+      stub.callsFake((...args) => {
+        if (args[0] === 'is:automation:client:connected') {
+          connectedCallback = args[2]
+        }
+
+        originalEmit.call(win.ws, ...args)
+      })
+    })
+
+    cy.get('[data-cy="spec-item"]').first().click()
+    // Let runner stabilize
+    cy.get('[data-cy="reporter-panel"]').should('be.visible').then(() => {
+      connectedCallback()
```

Comment:
Took me a bit to realize that by default, calling this without an argument will set the automation to disconnected. What about making it more explicit?

```js
connectedCallback(false)
```